### PR TITLE
Add picolibc

### PR DESCRIPTION
**Pull request by keith-packard**
Signed-off-by: Keith Packard <keithp@keithp.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,6 +12,9 @@ sdk-utilities:
 python:
 combo-package:
 
+.NOTPARALLEL:
```
Comment (cgsfv) [inline Makefile]: What does this do?

File: Makefile [hunk 2]
```diff
@@ -711,6 +716,7 @@ $(OBJDIR)/%/build/expat/configure:
 	rm -rf $(dir $@)
 	mkdir -p $(dir $@)
 	cp -a $(SRC_EXPAT)/* $(dir $@)
+	mkdir -p $(dir $@)/m4
```
Comment (cgsfv) [inline Makefile]: Why is this needed?

Comment (keith-packard) [reply Makefile]: It's obviously not needed for picolibc stuff, but when I tried to build locally, I got caught by a missing directory here. Probably just related to my local system configuration, but I didn't want to push anything to you that I hadn't tested.

File: Makefile (hunk 1 above)
Comment (keith-packard) [reply Makefile]: This disables the effect of my global MAKEFLAGS environment variable, which I have set to

export MAKEFLAGS=-j5

Without adding this to Makefile, the build would try to do things in parallel which needed to be serialized. I didn't spend time trying to figure out what additional dependencies would be needed to make parallel builds work correctly.

Comment (kito-cheng) [reply Makefile]: Problem is come from `make install`, and there is fix in @issuehsu 's PR https://github.com/sifive/freedom-tools/pull/24/commits/7b23d03c5ffa2e4e738e43dca9271e13b99b7f16#diff-b67911656ef5d18c4ae36cb6741b7965R785

Comment (keith-packard) [reply Makefile]: thanks for creating a fix that doesn't disable parallel builds in the entire tree.

**Comment by issuehsu** [comment]
Great! And I will rebase SEGGER libc support to latest branch.
